pd/src/state.rs

```diff
@@ -198,6 +201,15 @@ ON CONFLICT (id) DO UPDATE SET data = $1
             .await?;
         }
 
+        // Add undelegated note outputs into the quarantine queue.
+        for (pre_position, (note_commitment, quarantined_note)) in
+            block.undelegation_notes.into_iter().enumerate()
+        {
+            // TODO: quarantine undelegated note outputs
+        }
+
+        // TODO: quarantine nullifiers
+
```

Comment:
Should we fill this in before merging -- are there open questions on how we'd do that, or would it be straightforward?

Also, there should be a third step, which would be to add previously quarantined notes into the note commitment tree as they exit the unbonding period, right?

Reply:
This is now implemented.